[게시글 생성/조회/수정/삭제] 효오 미션 제출합니다.

hyojaekim opened the pull request:
안녕하세요.
새로운 것들을 배워나가면서 최대한 배운 내용을 적용하려고 하는데 궁금한게 있습니다.

Mapping이 많아지게 되면 Path를 한 곳에서 관리하는 것이 좋다고 생각되지만, 그대로 String으로 path를 입력하는 것도 직관적이라 각각의 장점이 있다고 생각합니다.

모든 path를 utils 패키지에 관리하는게 괜찮을까요? ㅎ

Deocksoo committed:
[DOCS] 해야될 일 목록을 업로드합니다

Deocksoo committed:
[DOCS] 기능 목록을 업데이트합니다

Deocksoo committed:
[ADD] 프로젝트에 필요한 의존성을 추가합니다

Deocksoo committed:
[ADD] Article과 ArticleRepository 도메인 객체를 추가합니다

Deocksoo committed:
[TEST] ArticleRepository 테스트에 주석을 추가합니다

Deocksoo committed:
[FIX] app.css 오류 코드를 수정합니다

Deocksoo committed:
[ADD] 게시글 작성 후 작성된 게시글을 보여주는 기능을 추가합니다

1. Article이 생성될 때 데이터를 저장하도록 수정하였습니다.

Deocksoo committed:
[ADD] IndexController를 추가합니다

index 페이지를 thymeleaf 템플릿 엔진에서 templates 디렉토리를 base로 하여 찾아줄 수
있도록 수정하였습니다.

Deocksoo committed:
[ADD] 메인 페이지 게시글 목록 조회 기능을 추가하였습니다

Deocksoo committed:
[ADD] 저장한 게시물을 조회하는 기능을 구현합니다

Deocksoo committed:
[REFACTOR] ArticleRepository find 메소드 리팩토링

Deocksoo committed:
[REFACTOR] article.html css, js 경로 수정하였습니다

Deocksoo committed:
[ADD] HellowWordController 실습 파일을 업로드합니다

Deocksoo committed:
[TEST] IndexController에서 사용하지 않는 라이브러리를 제거합니다

Deocksoo committed:
[ADD] 게시글 수정 및 해당 페이지로 이동하는 기능을 추가하였습니다

Deocksoo committed:
[REFACTOR] 컨벤션에 따른 수정 및 불필요한 라이브러리를 제거하였습니다

Deocksoo committed:
[ADD] 게시물 update 기능을 추가합니다

Deocksoo committed:
[ADD] 게시글 삭제 기능 구현 하였습니다

Deocksoo committed:
[REFACTOR] Article 클래스의 field 변수를 수정합니다

Deocksoo committed:
[REFACTOR] Custom exception class들을 추가합니다

Deocksoo committed:
[REFACTOR] ArticleController 메서드들의 이름과 순서를 변경합니다

Deocksoo committed:
[REFACTOR] Article.html파일의 script에서 initialValue를 수정합니다

Thymeleaf 템플릿 엔진을 통해 초기값을 지정해주도록 수정하였습니다.

Deocksoo committed:
[DOCS] 기능 목록을 업데이트합니다

Deocksoo committed:
[FIX] Custon Exception의 접근제어자를 변경합니다

Deocksoo committed:
[REFACTOR] Template 파일들을 리펙토링합니다

1. Template 파일들의 중복되는 코드들을 추출하였습니다.
2. 좌측 상단 배너를 클릭하면 index 페이지로 이동하도록 수정하였습니다.

Deocksoo committed:
[TEST] ArticleControllerTest의 articlePost메서드를 리펙토링 합니다

테스트에 사용할 지역변수들의 이름과 테스트 값을 변경하였습니다. input 값을 한글로
작성하는 경우 encoding 문제로 인해 성공하는 케이스가 실패하는 버그가 발생합니다.
이를 방지하기 위해 영어로 input 값들을 변경하였습니다.

Deocksoo committed:
[TEST] 몇몇 Test 클래스들의 어미를 Tests에서 Test로 변경합니다

Deocksoo committed:
[TEST] 불필요한 MyblogApplicationTest를 제거합니다

Deocksoo committed:
[REFACTOR] ArticleRepositoryTest를 리펙토링 합니다

1. 필드 상수명을 변경하였습니다.
2. 메서드 순서를 정리하였습니다.
3. 수정, 삭제할 때 id가 없는 경우에 대한 테스트 케이스들을 추가하였습니다.

Deocksoo committed:
[TEST] HelloWorldControllerTest에 DisplayName을 추가합니다

Deocksoo committed:
[REFACTOR] ArticleControllerTest를 리펙토링합니다

1. 필드 상수명을 변경하였습니다
2. 테스트 메서드명을 정리하였습니다.
3. 불필요한 테스트를 제거하였습니다.

Deocksoo committed:
[REFACTOR] ArticleRepository에서 Id를 할당하는 방식을 변경합니다

기존의 AutoIncrement 방식에서, 외부에서 id를 설정해 저장해주는 방식으로 변경하였습니다.

Deocksoo committed:
[ADD] ArticleController에서 수정, 삭제 후 redirect하도록 수정하였습니다

Deocksoo committed:
[FIX] index.html boarder 선이 스크롤 따라 움직이는 버그를 수정하였습니다

Deocksoo committed:
[ADD] 업데이트 된 css 파일을 추가합니다

hyojaekim committed:
[REFACTOR] AriticleController의 의존성을 받는 위치를 변경 하였습니다

hyojaekim committed:
[FIX] coverUrl 부분을 리팩토링 하였습니다

hyojaekim committed:
[REFACTOR] 나머지 html 파일의 경로를 templates 패키지로 이동하였습니다

hyojaekim committed:
[FIX] thymeleaf를 적용한 coverUrl에서 오타를 수정하였습니다

hyojaekim committed:
[REFACTOR] 사용하지 않는 라이브러리를 삭제합니다

hyojaekim committed:
[REFACTOR] 테스트의 내용의 이름을 수정합니다

phs1116 reviewed (changes requested):
안녕하세요! 리뷰어 화투입니다.
전체적으로 구현과 테스트 잘 해주셨어요 👍 
피드백 몇가지 추가했는데요, 참고 부탁드리고 궁금하신 점 있으시면 DM주세요!

phs1116 commented:
그리고 질문 사항에 대해서, Path는 별도 전역 객체가 관리하는것보다는
해당 컨트롤러에서만 사용하는 값이니 응집력있게 해당 컨트롤러가 책임지는게 적합합니다 :)

hyojaekim committed:
[REFACTOR] atomic을 적용해봤습니다

hyojaekim committed:
[REFACTOR] 객체에서 메세지를 보내서 확인하도록 수정하였습니다

hyojaekim committed:
[REFACTOR] 예외를 외부에서 처리하도록 수정하였습니다

hyojaekim committed:
[REFACTOR] 게시글 내용을 DTO로 전달하도록 수정하였습니다

hyojaekim committed:
[REFACTOR] ArticleRepository final로 수정하였습니다

hyojaekim committed:
[REFACTOR] IndexController 주입 방식을 변경하였습니다

기존에는 ArticleController에서 constructor injection, IndexController에서 field injection
모든 컨트롤러를 constructor injection 방식으로 통합하였습니다.

hyojaekim committed:
[REFACTOR] 테스트에서 요청하는 부분을 추상화 하였습니다

hyojaekim committed:
[REFACTOR] 요청하는 부분과 검증하는 부분을 분리하였습니다

pobiconan reviewed (approved):
피드백 반영하느라 수고했어요. 💯
